src/main/clojure/clara/rules/compiler.clj

```diff
@@ -2098,6 +2099,7 @@
                           (mapcat #(if (satisfies? IRuleSource %)
                                      (load-rules %)
                                      %))
+                          linter/validate-productions
```

Comment:
I thought this should be done in the compiler so that it would catch users providing rules outside the DSL but this could be checked there as well.  Thoughts?

Reply:
Would it have previously been broken to have duplicate bindings or is the addition of this linting a breaking change to existing rules?

I think typically I'd expect linting to be something you can opt in & out of.

That said, I think if it's going to happen somewhere in some automated way, the compiler makes sense to me - instead of dsl.

Reply:
Maybe "validation" would be a better name if we end up throwing an exception as here.  That seems more in keeping with checks for correctness as opposed to checks for questionable but valid code, which is what I'd expect from a linter.